specreduce/utils/utils.py

```diff
@@ -0,0 +1,181 @@
+    profile. `pixel` and `pixel_range` cannot be set simultaneously, and the default
+    is `pixel_range`=(min_image, max_image) to return an average profile across the
+    entire wavelength range of the image.
+
+    The window in the cross dispersion axis for measuring the profile at
+    the pixel(s) specified is determined by `width` and `trace`. If a trace is
+    provided (either as a Trace object, or as a number specifying the location
+    on the cross-dispersion axis of a FlatTrace object) that will determine the
+    center of the profile on the cross-dispersion axis. Otherwise, if `trace`
+    is None, the center of the image will be the center of the returned profile
+    (i.e., the center row assuming a horizontal trace).
+
+    If `width` is none, a window size of half the image in the cross-dispersion
+    axis will be used to measure the cross dispersion profile. Otherwise, an
+    integer value can be set for `width` which will determine the size of the
+    window around the trace used to measure the profile (this window moves with
+    the trace if trace is not flat).
+
+    Parameters
+    ----------
+    image : `~astropy.nddata.NDData`-like or array-like, required
+        2D image to measure cross-dispersion profile.
+    trace : Trace object, int, float, or None
+        A Trace object, a number to specify a FlatTrace, or None to use the
+        middle of the image. This is the position that defines the center of the
+        cross-dispersion profile. [default: None]
+    crossdisp_axis : int, optional
+        The index of the image's cross-dispersion axis. [default: 0]
+    width : tuple of int or None
+        Width around 'trace' to calculate profile. If None, then all rows in the cross-dispersion axis
+        will be used. [default: None]
+    pixel: int, list of int, or None
+        Pixel value(s) along the dispersion axis to return cross-dispersion profile.
+        If several specified in list, then the average (method set by `statistic`)
+        profile will be calculated. If None, and `pixel_range` is set, then
+        `pixel_range` will be used. [default: None]
+    pixel_range: tuple of int or None
+        Tuple of (min, max) defining the pixel range (along dispersion axis) to
+        calculate average cross-dispersion profile, up to and not inclusive of max.
+        If None, and `pixel` is not None, `pixel` will be used. If None and `pixel`
+        is also None, this will be interpreted as using the entire dispersion axis
+        to generate an average profile. [default: None]
+    statistic: 'median' or 'mean'
+        If `pixel` specifies multiple pixels, or `pixel_range` is specified, an
+        average profile will be returned. This can be either `median` (default)
+        or `mean`. This is ignored if only one pixel is specified. [default: median]
+    """
+
+    # work on copy of image so input isn't modified
+    image = copy.copy(image)
```

Comment:
would it be modified otherwise?  Or does either `_parse_image` or `MaskedArray` create a copy anyways?